generator/Data/XCB/Python/Parse.hs

```diff
@@ -413,6 +497,18 @@ mkPackStmts ext name m accessor prefix membs =
                                          (mkStr ('=' : packStr) : (map mkName args))]
       writeStmt = if length packStr > 0 then [StmtExpr write ()] else []
   in (args ++ listNames', writeStmt ++ listWrites)
+    where
+      mkListWrites :: [(Expr (), Maybe (Expr ()))]
+                   -> Statement()
+      mkListWrites [] = error "Empty list for packing"
+      mkListWrites [(expr, Nothing)] = flip StmtExpr () . mkCall "buf.write" $ (: []) expr
+      mkListWrites list = Conditional (map (uncurry mkConds) list) [] ()
```

Comment:
This set of cases looks a little funny to me (although I admit I haven't looked through it that closely); do you need mkListWrites seems like it's unecessary, since the singleton list element case looks very similar to the one in mkConds. Is there some way you can condense these? (This is all for style points, not that I think these are required to merge)